cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/AbstractResourceOperatorTest.java

```diff
@@ -106,7 +106,7 @@ public void createWhenExistsIsAPatch(TestContext context, boolean cascade) {
             }
             assertTrue(ar.succeeded());
             verify(mockResource).get();
-            verify(mockResource).patch(any());
+            //verify(mockResource).patch(any());
```

Comment:
Why is this necessary?

Reply:
`ConfigMapOperatorTest` runs this `createWhenExistsIsAPatch()` but current and desired CMs are equal. It expects the `patch()` to be called but nothing has changed and `patch()` is not called.

Reply:
Mmm, Perhaps it's always do the `patch()` in the `ConfigMapOperator.internalPatch()`, and evaluate whether the CM `data` changed after the call to the `super.internalPatch()`. The reason is that by not calling `patch()` we're not going to update the CM for other things than `data` like `metadata.labels`. Doing that would also mean you could reinstate this `verify()` call. It *would* mean that what we were returning from `ConfigMapOperator.internalPatch()` could sometimes be a no op when the patch wasn't actually a no op, but that's not a distinction that matters currently, so I think it would be acceptable.